twitter-finagle/src/main/scala-ce3/com/dwolla/util/async/finagle/ThriftClient.scala

```diff
@@ -16,13 +16,22 @@ object ThriftClient {
   def apply[Alg[_[_]] <: AnyRef {def asClosable: Closable}] = new PartiallyAppliedThriftClient[Alg]()
 
   class PartiallyAppliedThriftClient[Alg[_[_]] <: AnyRef {def asClosable: Closable} ] private[ThriftClient] (val dummy: Unit = ()) extends AnyVal {
-    def apply[G[_] : Async, F[_]](dest: String)
+    def apply[G[_] : Async, F[_]](dest: String, config: ThriftClientConfiguration = ThriftClientConfiguration())
                                  (implicit
                                   AFK: F ~~> G,
                                   AlgR: Alg[ReaderT[F, Alg[F], *]],
                                   CT: ClassTag[Alg[F]],
                                   FK: FunctorK[Alg]): Resource[G, Alg[G]] = {
-      val acquire = Sync[G].delay(Thrift.client.build[Alg[F]](dest)).map(_.asyncMapK[G])
+      val acquire =
+        Sync[G].delay(
+          Thrift
+            .client
+            .withSessionPool.maxSize(config.sessionPoolMax)
+            .withRequestTimeout(config.requestTimeout)
+            .withSession.acquisitionTimeout(config.requestTimeout)
+            .withRetryBudget(config.retryBudget)
```

Comment:
I think we want to be exhaustive here so we should also include a config for `.withTransport.connectTimeout()`

Reply:
Added